airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -79,10 +80,42 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
             return True, error_msg
         return True, None
 
+    def discover(self, logger: AirbyteLogger, config) -> AirbyteCatalog:
+        """
+        Override to provide the dynamic schema generation capabilities,
+        using resource available for authenticated user.
+
+        Retrieve: Bases, Tables from each Base, generate JSON Schema for each table.
+        """
+        auth = self._auth or BigqueryAuth(config)
+        # list all bases available for authenticated account
+        for dataset in BigqueryDatasets(project_id=config["project_id"], authenticator=auth).read_records(sync_mode=SyncMode.full_refresh):
+            dataset_id = dataset.get("datasetReference")["datasetId"]
```

Comment:
Based on the interface, `read_records` could return a `AirbyteMessage` and this code would not work. In practice, it seems to work though. Should we be explicit about this case and raise a specific exception saying that we don't expect that and there is a bug in the code if this error is shown?

Reply:
Hence can we add a case `if isinstance(dataset, AirbyteMessage): do x`?